presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -946,6 +951,21 @@ protected Type visitExists(ExistsPredicate node, StackableAstVisitorContext<Void
             return BOOLEAN;
         }
 
+        @Override
+        protected Type visitQuantifiedComparisonExpression(QuantifiedComparisonExpression node, StackableAstVisitorContext<Void> context)
```

Comment:
This should verify that the type of the expression implements the expected operators to be able to do the comparisons (orderable vs comparable)